### ISSUE TITLE
External links (target="_system") don't work on iOS 18

### ISSUE THREAD

**Issue by DagrutCr** [apache/cordova-ios#1511]
# Bug Report

## Problem

External links won't open anymore. I was on iOS 17 and it worked, I upgraded to iOS 18 and it does not anymore. Example : 

```html
<a target="_system" href="https://www.npmjs.com/package/cordova-ios">Click me</a>
```

### What is expected to happen?

The link should open in an external browser, most likely Safari.

### What does actually happen?

Nothing, no crash, log or message, except the one below in Xcode

## Information

Logs in Xcode when this happens :

```
BUG IN CLIENT OF UIKIT: The caller of UIApplication.openURL(_:) needs to migrate to the non-deprecated UIApplication.open(_:options:completionHandler:). Force returning false (NO).
```

### Command or Code

Build an app with the code below on iOS 18 and click the link : 

```html
<a target="_system" href="https://www.npmjs.com/package/cordova-ios">Click me</a>
```

### Environment, Platform, Device

- iPhone 14

### Version information

- iOS 18.1.1
- cordova-ios 7.1.1
- cordova 12.0.0

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

## Updates

Edit 1 : 
Further information about this : https://forums.developer.apple.com/forums/thread/763568
It seems that even the open() function fails in some cases.

Edit 2 : 
The issue does not seems to be from this repo directly. I keep searching...

**Comment by dpogue** [comment]
hmmm, I wonder if that's coming from here: https://github.com/apache/cordova-ios/blob/9beca6d7d3368c157cb5f712e348860dc0aceb3e/CordovaLib/Classes/Private/Plugins/CDVIntentAndNavigationFilter/CDVIntentAndNavigationFilter.m#L130

We can probably change that to call `open(_:options:completionHandler:)` instead

Just for the record, I don't believe `target="_system"` actually has any special meaning when used on links.

**Comment by DagrutCr** [comment]
That's not the faulty line here, after further investigations this line is not even called in my case (I've added some `NSLog()` before and after, none were in the logs). I guess it's from a plugin. That plugin may use a similar call though, so I'll see and report back what I find here :-)

I'll keep looking, I should have an answer today.

**Comment by DagrutCr** [comment]
I found where it was from : https://github.com/apache/cordova-plugin-inappbrowser/blob/b3a0d82c60ee2cff2d2b4ca8160325c34bd7b887/src/ios/CDVWKInAppBrowser.m#L364

The issue arise when we are calling `openUrl` with no extra parameters, which is not the case here. I have some other plugins with this issue, I see. I'll report them or fix them.

Thanks! :-)

**Comment by kenguest** [comment]
hi @DagrutCr @dpogue I don't understand why this ticket is marked as completed when the issue is still present in the code? (the problem is still happening that external links won't open on devices running iOS 18, and I don't see any new releases for the inappbrowser project)

Do you need an issue/ticket created in the cordova-plugin-inappbrowser project instead? If that's the case, I've created one at https://github.com/apache/cordova-plugin-inappbrowser/issues/1080 with some more info in it than in this ticket.

Thanks - and please let me know if you need me to provide any more information if needed!

**Comment by DagrutCr** [comment]
Hi @kenguest !

The issue is not from this plugin, nor from cordova-plugin-inappbrowser (I was not using the latest version). I switched to the git version if that may help, though. The plugin causing the issue for me was cordova-plugin-photo-library, and I fixed it in the last two commits here : https://github.com/DagrutCr/cordova-plugin-photo-library/commits/master/ .

**Comment by kenguest** [comment]
hi @DagrutCr !
Thanks for responding so quick!
I'm not using that cordova-plugin-photo-library plugin - I guess the same bug is present in one of the other plugins that I'm using!

Thanks again!

**Comment by AlexDonn19** [comment]
See this link:
https://discussions.unity.com/t/application-openurl-doesnt-work-on-ios-18/1553960/12
You can change CDVWKInAppBrowser.m , string 364 to something like this:
// [[UIApplication sharedApplication] openURL:url];
if (@available(iOS 10.0, *)) {
 [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:^(BOOL success) {
 if (!success) {
  NSLog(@"Failed to open URL: %@", url);
 }
}];
} else {
 BOOL success = [[UIApplication sharedApplication] openURL:url];
 if (!success) {
  NSLog(@"Failed to open URL: %@", url);
 }
}
